### PR TITLE
TMA-710: segments filter in user filters and users bricks

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 1]
```diff
@@ -167,18 +167,18 @@ def call(params)
             end
 
             domain_clients = domain.clients(:all, data_product)
-            if params.segments_filter
-              segment_uris = params.segments_filter.map(&:uri)
-              domain_clients.select! { |c| segment_uris.include?(c.segment_uri) }
+            if params.segments
+              segment_uris = params.segments.map(&:uri)
+              domain_clients = domain_clients.select { |c| segment_uris.include?(c.segment_uri) }
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'c.segment_uri' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'segment_uris.include?(c.segment_uri)' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Has the variable name 'c' (UncommunicativeVariableName)

File: lib/gooddata/lcm/actions/synchronize_user_filters.rb [hunk 2]
```diff
@@ -167,18 +167,18 @@ def call(params)
             end
 
             domain_clients = domain.clients(:all, data_product)
-            if params.segments_filter
-              segment_uris = params.segments_filter.map(&:uri)
-              domain_clients.select! { |c| segment_uris.include?(c.segment_uri) }
+            if params.segments
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_user_filters.rb]: Calls 'params.segments' 4 times (DuplicateMethodCall)

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 3]
```diff
@@ -168,17 +168,17 @@ def call(params)
...
                         domain_clients.select! { |c| segment_uris.include?(c.segment_uri) }
                       end
                       working_client_ids = []
                       res = []
                       res += new_users.group_by { |u| u[:pid] }.flat_map do |client_id, users|
                         fail "Client id cannot be empty" if client_id.blank?
                         c = domain.clients(client_id, data_product)
-                        if params.segments_filter && !segment_uris.include?(c.segment_uri)
-                          puts "Client #{client_id} is outside segments_filter #{params.segments_filter}"
+                        if params.segments && !segment_uris.include?(c.segment_uri)
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'c.segment_uri' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'segment_uris.include?(c.segment_uri)' 2 times (DuplicateMethodCall)

**Comment by panjan** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/597/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1139/ : SUCCESS in 49m 52s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/897/ : SUCCESS in 28m 28s

**Pull request merged**